model/task/task.go

```diff
@@ -3448,6 +3452,21 @@ func (t *Task) SetDisplayTaskID(id string) error {
 		}}))
 }
 
+// SetCheckRunId sets the checkRunId for the task
+func (t *Task) SetCheckRunId(checkRunId int) error {
+	t.CheckRunId = utility.ToIntPtr(checkRunId)
+	return UpdateOne(
```

Comment:
It's slightly more preferable to `UpdateOne` before setting the field in memory:
```
if err := UpdateOne(<update>...); err != nil {
    return err
}

t.CheckRunId = checkRunId
return nil
```
This avoids a general issue where the task in-memory (i.e. `t`) has a different state from the DB if the update fails. It's caused some bugs in the past where an update failed but `t` was updated, so the logic in the outer function acted on wrong information.